### PR TITLE
15 create per experiment schedulers

### PR DESCRIPTION
**Pull request by recap**
Switching to per experiment pluggable schedulers. 

- Each experiment defines its own scheduler in config.json
- The schedulers are loaded from a directory on boot up
- Schedulers manage the queuing

### CODE REVIEW

File: class_loader.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+// Class for dynamically loading class plugins. 
+// The inner function recursiveWalk, recursively 
+// walks a directory and load classes to classMap 
+// Map
+class ClassLoader {
+  constructor(classPath) {
+    this.classMap = new Map()
+    this.classPath = classPath
+    this.#recursiveWalk(this.classPath)
```
Comment (recap) [inline class_loader.js]: check for out of order execution

Comment (ewan-escience) [reply class_loader.js]: You can write a static async factory method instead: https://dev.to/somedood/the-proper-way-to-write-async-constructors-in-javascript-1o8c

File: server.js [hunk 2]
```diff
@@ -142,17 +144,19 @@ const validateToken = (req, res, next) => {
...
    *
    * @type {Map<string, User>}
    */
-  const usersDb = new Map()
+  const usersDb = new Map() 
   const userMapping = {}
   const agreementIds = {}
   const expToEnable = config.experiments.map(e => e.name)
+  // Load schedulers from directory
+  const SchedulerPluggins = new ClassLoader('./schedulers')
```
Comment (ewan-escience) [inline server.js]: Spelling: Pluggins -> Plugins

File: local-queue.js [hunk 3]
```diff
@@ -1,65 +1,42 @@
...
-
-// Get or create queue
-function getOrCreateQueue(queueName) {
-  if (!queues.has(queueName)) {
-    queues.set(queueName, new Array())
+// A LocalQueue implementation
+class LocalQueue {
+  constructor(queueName) {
+    this.name = queueName
+    this.queue = new Array()
```
Comment (ewan-escience) [inline local-queue.js:5]: Keep in mind for the future: also have a Set for quick `has` operations. Furthermore, you might have to make the methods `async` again to be consistent with the Redis queue in the future.

**Review by ewan-escience — CHANGES_REQUESTED** [review]
Nice work! Some small changes and remarks to consider later.

File: class_loader.js (hunk 1 above)
Comment (recap) [reply class_loader.js]: fixed with static asyn initialize() function

File: local-queue.js (hunk 3 above)
Comment (recap) [reply local-queue.js:5]: created an issue #39

**Review by ewan-escience — APPROVED** [review]

**Pull request merged**